ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts

```diff
@@ -1808,7 +1891,668 @@ const triggerSet: TriggerSet<Data> = {
+        stack: Outputs.getTogether,
+        unknown: Outputs.unknown,
+        ...Directions.outputStrings8Dir,
+      },
+    },
+    {
+      id: 'FRU P4 Crystallize Time Blue Cleanse',
+      type: 'Ability',
+      netRegex: { id: '9D55', capture: false }, // Unholy Darkness
+      condition: (data) => data.phase === 'p4-ct',
+      delaySeconds: 2,
+      durationSeconds: 8,
+      suppressSeconds: 1,
+      infoText: (data, _matches, output) => {
+        const debuff = data.p4CTMyRole;
+        if (!debuff)
+          return;
+
+        if (debuff.startsWith('blue'))
+          return output.cleanse!({ spot: output[debuff]!() });
+        return output.avoidCleanse!();
+      },
+      outputStrings: {
+        cleanse: {
+          en: 'Cleanse: ${spot}',
+        },
+        avoidCleanse: {
+          en: 'Avoid cleanse puddles',
+        },
+        blueWater: Directions.outputStrings8Dir.dirSE!,
+        blueUnholy: Directions.outputStrings8Dir.dirE!,
+        blueEruption: Directions.outputStrings8Dir.dirW!,
+        blueIce: Directions.outputStrings8Dir.dirSW!,
+      },
+    },
+    {
+      id: 'FRU P4 Crystallize Time Tidal Light Collect',
+      type: 'StartsUsing',
+      netRegex: { id: '9D3B' }, // Tidal Light
+      condition: (data) => data.phase === 'p4-ct',
+      run: (data, matches) => {
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+        const dirNum = Directions.xyTo8DirNum(x, y, centerX, centerY);
+        data.p4CTTidalDirs.push(dirNum);
+      },
+    },
+    {
+      id: 'FRU P4 Crystallize Time Drop Rewind',
+      type: 'GainsEffect',
+      netRegex: { effectId: '1070' }, // Spell-in-Waiting: Return
+      condition: (data, matches) => data.phase === 'p4-ct' && data.me === matches.target,
+      delaySeconds: (_data, matches) => parseFloat(matches.duration) - 8,
+      countdownSeconds: 8,
+      alertText: (data, _matches, output) => {
+        const unknownStr = output.rewind!({ spot: output.unknown!() });
+        if (data.p4CTTidalDirs.length !== 2)
+          return unknownStr;
+
+        // re-use findURNorthDirNum() to find the intercard between the two starting Tidal spots
+        const intersectDirNum = findURNorthDirNum(data.p4CTTidalDirs);
+        if (intersectDirNum === -1)
+          return unknownStr;
+
+        const dir = Directions.output8Dir[intersectDirNum];
+        if (dir === undefined)
+          return unknownStr;
+
+        return output.rewind!({ spot: output[dir]!() });
+      },
+      outputStrings: {
+        rewind: {
+          en: 'Drop Rewind: ${spot}',
+        },
+        unknown: Outputs.unknown,
+        ...Directions.outputStringsIntercardDir,
+      },
+    },
+    {
+      id: 'FRU P4 Crystallize Time Spirit Taker',
+      type: 'GainsEffect',
+      netRegex: { effectId: '994' }, // Return
+      condition: (data, matches) => data.phase === 'p4-ct' && data.me === matches.target,
+      response: Responses.spread('alert'),
```

Comment:
This one's probably worth doing a special-cased trigger as `Spread -- Avoid crystal`, since it's easy to forget that when panicking after doing the rest of the mechanic.